components/signup/signup_email/signup_email.tsx

```diff
@@ -41,16 +53,20 @@ export default class SignupEmail extends React.PureComponent {
         }).isRequired,
     }
 
-    constructor(props) {
+    private emailRef: React.RefObject<HTMLInputElement>;
+    private nameRef: React.RefObject<HTMLInputElement>;
+    private passwordRef: React.RefObject<HTMLInputElement>;
+
+    constructor(props: Props) {
         super(props);
 
         const data = (new URLSearchParams(this.props.location.search)).get('d');
         const token = (new URLSearchParams(this.props.location.search)).get('t');
         const inviteId = (new URLSearchParams(this.props.location.search)).get('id');
 
-        this.state = {};
+        this.state = {} as State;
```

Comment:
Since the only non-nullable field in the state is `loading`, it might be better to have `this.state = {loading: false};` here instead of using a type assertion.

Reply:
Second this ☝️